src/material/core/tokens/m2/_index.scss

```diff
@@ -0,0 +1,93 @@
+@use 'sass:list';
+@use 'sass:map';
+@use 'sass:meta';
+@use '../../theming/palette';
+@use '../../theming/theming' as theming;
+@use '../../typography/all-typography';
+@use '../../style/sass-utils';
+
+@use './mat/card' as tokens-mat-card;
+@use './mat/radio' as tokens-mat-radio;
+@use './mat/snack-bar' as tokens-mat-snack-bar;
+@use './mat/tab-header' as tokens-mat-tab-header;
+@use './mat/tab-header-with-background' as tokens-mat-tab-header-with-background;
+@use './mdc/checkbox' as tokens-mdc-checkbox;
+@use './mdc/circular-progress' as tokens-mdc-circular-progress;
+@use './mdc/dialog' as tokens-mdc-dialog;
+@use './mdc/elevated-card' as tokens-mdc-elevated-card;
+@use './mdc/icon-button' as tokens-mdc-icon-button;
+@use './mdc/linear-progress' as tokens-mdc-linear-progress;
+@use './mdc/list' as tokens-mdc-list;
+@use './mdc/outlined-card' as tokens-mdc-outlined-card;
+@use './mdc/radio' as tokens-mdc-radio;
+@use './mdc/snack-bar' as tokens-mdc-snack-bar;
+@use './mdc/tab' as tokens-mdc-tab;
+@use './mdc/tab-indicator' as tokens-mdc-tab-indicator;
+
+// Option to save the original theme object into the tokens map, so that theme mixins can continue
+// using it while migration to tokens is in progress.
+$_enable-back-compat-theme-access: false;
+
+// Gets the tokens for the given theme, m2 tokens module, and theming system.
+// Valid theming systems are: unthemable, color, typography, density.
+@function _get-tokens-for-module-and-system($theme, $module, $system) {
+  @if $system == unthemable {
+    @return meta.call(
+        meta.get-function(get-#{$system}-tokens, $module: $module));
+  }
+  @return meta.call(
+      meta.get-function(get-#{$system}-tokens, $module: $module), map.get($theme, $system));
+}
+
+// Gets the fully qualified tokens map for the given theme and m2 tokens module.
+@function _get-tokens-for-module($theme, $module) {
+  $tokens: sass-utils.deep-merge-all(
+      _get-tokens-for-module-and-system($theme, $module, unthemable),
+      _get-tokens-for-module-and-system($theme, $module, color),
+      _get-tokens-for-module-and-system($theme, $module, typography),
+      _get-tokens-for-module-and-system($theme, $module, density));
+  @return map.set((), map.get(meta.module-variables($module), prefix), $tokens);
+}
+
+// Gets the full set of m2 tokens for the given theme object. If parts of the theme are missing
+// (color, density, etc), defaults are used. Returned format:
+// (
+//   (fully, qualified, namespace): (
+//     token: value
+//   )
+// )
+@function m2-tokens-from-theme($theme: null) {
```

Comment:
Would it make sense to define this in a file like `_tokens-backwards-compat.scss`?